velox/dwio/dwrf/common/FileMetadata.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/dwio/dwrf/common/FileMetadata.h"
+
+namespace facebook::velox::dwrf {
+namespace detail {
+using dwio::common::CompressionKind;
+
+CompressionKind orcCompressionToCompressionKind(
+    proto::orc::CompressionKind compression) {
+  switch (compression) {
+    case proto::orc::CompressionKind::NONE:
+      return CompressionKind::CompressionKind_NONE;
+    case proto::orc::CompressionKind::ZLIB:
+      return CompressionKind::CompressionKind_ZLIB;
+    case proto::orc::CompressionKind::SNAPPY:
+      return CompressionKind::CompressionKind_SNAPPY;
+    case proto::orc::CompressionKind::LZO:
+      return CompressionKind::CompressionKind_LZO;
+    case proto::orc::CompressionKind::LZ4:
+      return CompressionKind::CompressionKind_ZSTD;
+    case proto::orc::CompressionKind::ZSTD:
+      return CompressionKind::CompressionKind_LZ4;
+  }
+  return CompressionKind::CompressionKind_NONE;
+}
+} // namespace detail
+
+PostScript::PostScript(const proto::PostScript& ps)
+    : footerLength_{ps.footerlength()},
+      compression_{
+          ps.has_compression()
```

Comment:
I was also planning to do the same but I found that many tests use `PostScript` expecting that it copies values of `proto::PostScript`. Like
https://github.com/facebookincubator/velox/blob/ff331272ad5f67ad5585085573c3401528282de1/velox/dwio/dwrf/test/ReaderBaseTests.cpp#L100
It sets up a temporary `proto::PostScript` and use it to build a `ReaderBase` member. As the pointer method makes `PostScript` not owning the object, tests like this will not work.
Maybe we should change the tests together?

Reply:
I see, this is an object we will take ownership.  You can make a wrapper like this:
```c++
struct PostScript {
  std::unique_ptr<google::protobuf::Message> impl_;
  DwrfFormat format_;
};
```